src/build_queue.rs

```diff
@@ -266,8 +307,15 @@ impl BuildQueue {
             }
         }
 
+        // store the last seen reference as git reference in
+        // the local crates.io index repo.
         diff.set_last_seen_reference(oid)?;
 
+        // additionally set the reference in the database
+        // so this survives recreating the registry watcher
+        // server.
+        self.set_last_seen_reference(oid)?;
```

Comment:
It worries me that this isn't atomic. I worry that we'll:
- update this in the local git repo
- fail to update it in the database
- end up building the same list of crates twice

I guess that's not the end of the world. But it would be nice to add some more error logging, and to update it in the database before updating the git repo.

Reply:
I changed the order of these, first setting the ref in the db, then in the repo. 

What additional logging to you imagine? 

At the call-site of `get_new_crates` ( now in `watch_registry`) all errors will be logged already.